packages/grid/x-grid/src/tests/editRows.XGrid.test.tsx

```diff
@@ -294,4 +371,50 @@ describe('<XGrid /> - Edit Rows', () => {
       },
     });
   });
+
+  it('should stay in the edit mode when clicking in an element inside a portal', () => {
+    render(
+      <TestCase
+        columns={[
+          {
+            field: 'brand',
+            editable: true,
+            renderEditCell: () => (
+              <Portal>
+                <button>Click me</button>
+              </Portal>
+            ),
+          },
+        ]}
+      />,
+    );
+    const cell = getCell(0, 0);
+    expect(cell).not.to.have.class('MuiDataGrid-cellEditing');
+    fireEvent.doubleClick(cell);
+    expect(cell).to.have.class('MuiDataGrid-cellEditing');
+    fireEvent.mouseUp(screen.getByRole('button', { name: /Click me/i }));
+    fireEvent.click(screen.getByRole('button', { name: /Click me/i }));
+    expect(cell).to.have.class('MuiDataGrid-cellEditing');
+  });
+
+  it('should stay in the edit mode when the element inside the cell triggers click but no mouseup', () => {
+    render(
+      <TestCase
+        columns={[
+          {
+            field: 'brand',
+            editable: true,
+            renderEditCell: () => <input type="checkbox" />,
+          },
+        ]}
+      />,
+    );
+    const cell = getCell(0, 0);
+    expect(cell).not.to.have.class('MuiDataGrid-cellEditing');
+    fireEvent.doubleClick(cell);
+    expect(cell).to.have.class('MuiDataGrid-cellEditing');
+    const checkbox = screen.getByRole('checkbox', { key: ' ' });
```

Comment:
?

```suggestion
    const checkbox = screen.getByRole('checkbox');
```

Reply:
I forgot to remove the `key`. My initial idea was to test pressing <kbd>Space</kbd> on a checkbox, because doing that triggers a `click` but not a `mouseup`. See https://bugzilla.mozilla.org/show_bug.cgi?id=1120106